examples/parent_child.py

```diff
@@ -0,0 +1,168 @@
+    """
+    author = Object(User)
+    created = Date()
+    content = Text()
+
+class Post(DocType):
+    """
+    Base class for Question and Answer containing the common fields.
+    """
+    author = Object(User)
+    created = Date()
+    body = Text()
+    comments = Nested(Comment)
+    question_answer = Join(relations={'question': 'answer'})
+
+    def save(self, **kwargs):
+        # if there is no date, use now
+        if self.created is None:
+            self.created = datetime.now()
+        return super(Post, self).save(**kwargs)
+
+class Question(Post):
+    # use multi True so that .tags will return empty list if not present
+    tags = Keyword(multi=True)
+    title = Text(fields={'keyword': Keyword()})
+
+    def add_answer(self, user, content, created=None, accepted=False):
+        Answer(
+            # required make sure the answer is stored in the same shard
+            _routing=self.meta.id,
+            # since we don't have explicit index, ensure same index as self
+            _index=self.meta.index,
+            # set up the parent/child mapping
+            question_answer={'name': 'answer', 'parent': self.meta.id},
+
+            # pass in the field values
+            author=user,
+            created=created,
+            content=content,
+            accepted=accepted
+        ).save()
+
+    def search_answers(self):
+        # search only our index
+        s = self.search(index=self.meta.index)
+        # filter for answers belonging to us
+        s = s.filter('parent_id', type="answer", id=self.meta.id)
+        # add routing to only go to specific shard
+        s = s.params(routing=self.meta.id)
+        return s
+
+    class Meta:
+        def matches(self, hit):
+            " Use Question class for parent documents "
+            return hit.question_answer == 'question'
+
+
+class Answer(Post):
+    is_accepted = Boolean()
+
+    @property
+    def question(self):
+        # cache question in self.meta
+        # any attributes set on self would be interpretted as fields
+        if 'question' not in self.meta:
+            self.meta.question = Question.get(
+                    id=self.question_answer.parent, index=self.meta.index)
+        return self.meta.question
+
+    class Meta:
+        def matches(self, hit):
+            " Use Answer class for child documents with child name 'answer' "
+            return getattr(hit.question_answer, 'name', None) == 'answer'
+
+
+# construct basic index and populate with default settings. This will then be
+# used to create a template
+BASE_INDEX = Index('*.stackexchange.org')
+BASE_INDEX.settings(
+    number_of_shards=1,
+    number_of_replicas=0
+)
+# add the DocType classes to the index, their mappings will be merged
+BASE_INDEX.doc_type(Answer)
+BASE_INDEX.doc_type(Question)
+
+
+def setup():
+    " Create an IndexTemplate and save it into elasticsearch. "
+    index_template = BASE_INDEX.as_template('base')
```

Comment:
`as_template` exists?

Reply:
ooops, it exists in `meta-index` branch, part of #840 

I got my branches crossed, sorry about that